### PR TITLE
docs(sdk): simpler documentation for building the sdk locally

### PR DESCRIPTION
**Pull request by npretto**
This pr fixes some issues we had in the dev docs + adds new parts about common errors.

It also removes the sdk dev docs from the docs website to keep them "internal" for the moment as discussed yesterday in the weekly

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/dev.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Storybook expects an instance running on `localhost:3000` with some configuration:
+
+- on [/admin/settings/authentication/jwt](http://localhost:3000/admin/settings/authentication/jwt)
+  - Make sure "User Provisioning" is enabled.
+  - Set JWT secret to be "`0000000000000000000000000000000000000000000000000000000000000000`" in Admin > Authentication >
+    JWT > String used by the JWT signing key
+  - You will need to input a `JWT Identity Provider URI` even though it will not be used.
+  - Press "Save and enable"
+- on [/admin/settings/embedding-in-other-applications](http://localhost:3000/admin/settings/embedding-in-other-applications)
+  - Enable the sdk
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/dev.md:35]: we previously had instructions to add * as authorized origins for interactive embedding, but we don't need it anymore with this new flag

File: enterprise/frontend/src/embedding-sdk/test/CommonSdkStoryWrapper.tsx [hunk 2]
```diff
@@ -37,6 +37,7 @@ export const storybookSdkDefaultConfig: SDKConfig = {
       return response.json();
     } catch (e) {
       console.error("Failed to generate JWT", e);
+      return `Failed to generate JWT for storybook: ${e}`;
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/test/CommonSdkStoryWrapper.tsx:40]: This is the wrapper used in storybook, the returned string will be rendered in the error component in the `The ${source} must return an object with the shape {id:string, exp:number, iat:number, status:string}, got ${safeStringify(session)} instead` message.
This should make it a bit easier to debug, as previously we didn't return anything

File: enterprise/frontend/src/embedding-sdk/dev.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- install the sdk with `yarn add file:../metabase/resources/embedding-sdk` / `npm install --install-links ../metabase/resources/embedding-sdk`
+- clean the cache of your bundler
+
+An example for vite: `rm -rf node_modules/@metabase && rm -rf node_modules/.vite && yarn add file:../metabase/resources/embedding-sdk`
+
+We usually create a custom `dev:link` script in the package.json to simplify the process.
+
+#### `Cannot read properties of null (reading 'useRef')`
+
+This happens when multiple versions of react are found in the project.
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/dev.md:91]: This is my understanding of the problem at least, it messes up with symlinks and with npm projects nested in other projects

File: docs/developers-guide/sdk.md [hunk 4]
```diff
@@ -1,77 +0,0 @@
...
-
-```bash
-yarn build-embedding-sdk:watch
-```
-`build-embedding-sdk:watch` is the original command, the js output is fast, but the dts output is extremely slow and is not fixed by the fixup script on watch.
-
-## Build incrementally
-
-```bash
-yarn build-embedding-sdk:dev
```
Comment (npretto) [inline docs/developers-guide/sdk.md:29]: This doesn't exist

File: docs/developers-guide/sdk.md [hunk 5]
```diff
@@ -1,77 +0,0 @@
...
-```bash
-yarn storybook-embedding-sdk
-```
-
-## Initial configuration
-
-1. Set JWT secret to be "`0000000000000000000000000000000000000000000000000000000000000000`" in Admin > Authentication >
-   JWT > String used by the JWT signing key
-1. Make sure "User Provisioning" setting is set to "`on`".
-1. Set Authorized Origins to "`*`" in Admin > Embedding > Interactive embedding
```
Comment (npretto) [inline docs/developers-guide/sdk.md:52]: Not true anymore

File: enterprise/frontend/src/embedding-sdk/dev.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+Storybook is a quick way of testing sdk components with hot reloading.
+
+Storybook expects an instance running on `localhost:3000` with some configuration:
+
+- on [/admin/settings/authentication/jwt](http://localhost:3000/admin/settings/authentication/jwt)
+  - Make sure "User Provisioning" is enabled.
+  - Set JWT secret to be "`0000000000000000000000000000000000000000000000000000000000000000`" in Admin > Authentication >
+    JWT > String used by the JWT signing key
+  - You will need to input a `JWT Identity Provider URI` even though it will not be used.
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/dev.md:34]: I think this was missing from the previous docs, and it's still a bit confusing that we need to do it so i added the step

File: enterprise/frontend/src/embedding-sdk/dev.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+# Docs for development on the sdk
+
+These docs are for building the sdk locally, if you just want to use the sdk, please refer to the [sdk documentation](https://www.metabase.com/docs/latest/embedding/sdk/introduction).
+
+## Build
+
+You can build the sdk with `build-embedding-sdk`.
```
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/dev.md:7]: I assume these are yarn commands, maybe we prefix all of them with `yarn`?

File: enterprise/frontend/src/embedding-sdk/dev.md (hunk 3 above)
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/dev.md:91]: Maybe we should include that symlink explanation so people might have some more context about it

**Review by oisincoveney — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/dev.md (hunk 3 above)
Comment (npretto) [reply enterprise/frontend/src/embedding-sdk/dev.md:91]: I included some more context, does it check out now?

File: enterprise/frontend/src/embedding-sdk/dev.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+Storybook will use the source files and not the built package.
+
+## E2E tests
+
+E2e tests for the sdk are currently done with cypress component tests.
+
+```bash
+yarn test-cypress-open-component-sdk
+```
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/dev.md:49]: Optional: we could add the envs that are needed to run the EE tests to the docs:

```
MB_EDITION=ee
CYPRESS_ALL_FEATURES_TOKEN=
CYPRESS_NO_FEATURES_TOKEN=
```

**Review by heypoom — APPROVED** [review]
Very nice! It should be possible for non-embedding folks to run the SDK by themselves now 🎉

File: enterprise/frontend/src/embedding-sdk/dev.md (hunk 8 above)
Comment (npretto) [reply enterprise/frontend/src/embedding-sdk/dev.md:59]: Good point! I always forget about it as I have them globally
b5c669d3072

**Review by oisincoveney — APPROVED** [review]

**Comment by npretto** [comment]
@metabase-bot backport release-x.51.x

**Comment by npretto** [comment]
@metabase-bot backport release-x.51.x

**Comment by npretto** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**